lib/bowling_score.rb

```diff
@@ -0,0 +1,55 @@
+class BowlingScore
+  attr_reader :total_score, :frame_number, :current_frame, :frames
+
+  def initialize
+    @total_score = 0
+    @frame_number = 1
+    @current_frame = []
+    @frames = {}
+  end
+
+  def roll(pins)
+    @current_frame << pins
+    @total_score += pins * number_of_bonuses
+    if @current_frame.length == 2 || pins == 10
```

Comment:
This is generally a good one, but it wont allow for the tenth frame to have a third roll, how would you approach that?

Reply:
Yes, I hadn't gotten quite that far! I thought I would probably define a separate predicate method (e.g. `tenth_frame?`) and add a conditional line of `if tenth_frame?` then execute something different (probably a separate method, or have a parent method calling either this or the tenth frame method as appropriate).